src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1284,12 +1289,7 @@ public Type[] getGenericInterfaces() {
      * @since 1.1
      */
     public Class<?> getComponentType() {
-        // Only return for array types. Storage may be reused for Class for instance types.
```

Comment:
I don't see any changes to componentType related to reuse.  So was this comment and the code below already obsolete?

Reply:
It was. Before the componentType field was reused for the class initialization monitor int array, and it caused problems with core reflection if a program reflectively accesses this field after a few hundred times. See [JDK-8337622](https://bugs.openjdk.org/browse/JDK-8337622).

Reply:
So for JDK 8 and 21+, the init_lock and componentType are not shared.  In JDK 11 and 17, Hotspot shares the fields, but it's not observable with the older implementation of reflection.  See https://bugs.openjdk.org/browse/JDK-8337622.